Modules/rats.py

```diff
@@ -162,15 +183,45 @@ def get_rat(cls, name: str, platform: Platforms = Platforms.DEFAULT) -> 'Rats' o
         if not isinstance(name, str) or not isinstance(platform, Platforms):
             raise TypeError("invalid types given.")
 
+        # initialize before use
+        found = None
+
         try:
             found = Rats.cache_by_name[name]
         except KeyError:
             # no such rat in cache
-            return None
+            if cls.apiHandler is not None:
+                found = await cls.apiHandler.someApiCall(name=name)  # pragma: no cover
+                # FIXME: replace SomeApiCall with the actual call once we have the interface
+            return found
         else:
             # we found a rat
             return found if (found.platform == platform or platform == Platforms.DEFAULT) else None
 
+    @classmethod
+    async def get_rat_by_uuid(cls, uuid: UUID) -> 'Rats' or None:
+        """
+        Finds a rat by their UUID.
+
+        This method will first check the local cache and, in the event of a cache miss, will make an API call.
+
+        Args:
+            uuid (UUID): api uuid to find a rat for
+
+        Returns:
+            Rats: found Rescue
+        """
+        if not isinstance(uuid, UUID):
+            raise TypeError
+        else:
+            found = None
+            if uuid in cls.cache_by_id:
+                found = cls.cache_by_id[uuid]
+            elif cls.apiHandler is not None:  # pragma: no cover
+                found = await cls.apiHandler.someApiCall(uuid=uuid)
```

Comment:
I can tell you right now that the signature is `async def get_rat_by_id(self, id: Union[str, UUID]) -> Rats`